src/runtime/BorrowedReference.cs

```diff
@@ -10,8 +10,16 @@ namespace Python.Runtime
         readonly IntPtr pointer;
         public bool IsNull => this.pointer == IntPtr.Zero;
 
+
         /// <summary>Gets a raw pointer to the Python object</summary>
-        public IntPtr DangerousGetAddress() => this.pointer;
+        public IntPtr DangerousGetAddress()
+            => this.IsNull ? throw new NullReferenceException() : this.pointer;
+
+        /// <summary>
+        /// Gets a raw pointer to the Python object. Does not throw an exception
+        ///  if the pointer is null
+        /// </summary>
+        public IntPtr DangerousGetAddressUnchecked() => this.pointer;
```

Comment:
NIT: I don't think you are actually using this anywhere

Reply:
Indeed, it's not used anymore